main.go

```diff
@@ -55,6 +105,52 @@ func transform(in io.Reader, out io.Writer, C int, B bool) error {
 	return nil
 }
 
+func ReverseHandler(w http.ResponseWriter, r *http.Request) {
+	result, err := NewReverseTransformer().Transform(r.Body)
+	if err != nil {
+		http.Error(w, "Server ReverseHandler Transformer error", http.StatusBadRequest)
```

Comment:
Send correct http response codes: in this case, it looks more like Server Internal Error, but definitely not Bad Request